src/runtime/util.cpp

```diff
@@ -39,32 +39,32 @@ void parseSlice(BoxedSlice* slice, int size, i64* out_start, i64* out_stop, i64*
     int64_t istop;
     int64_t istep = 1;
 
-    if (isSubclass(step->cls, int_cls)) {
+    if (step->cls != none_cls && (step->cls == int_cls || isSubclass(step->cls, int_cls))) {
```

Comment:
ah I'd added that just to reduce calls to isSubclass (which were rather hot in `make perf` output while I was working on this.)  given @undingen's changes this should be unnecessary.